pennylane/beta/tapes/qnode.py

```diff
@@ -137,6 +144,25 @@ def __init__(self, func, device, interface="autograd", diff_method="best", **dif
         self.dtype = np.float64
         self.max_expansion = 2
 
+        self._caching = None
+        """float: number of device executions to store in a cache to speed up subsequent
+        executions. If set to zero, no caching occurs."""
+
+        if caching is not None:

```

Comment:
Could the default for the `caching` argument be `0`? Also, it seems that if the user specified `0`, then this condition would evaluate to `True`, correct? :thinking: